src/Bindings/PluginLua.h

```diff
@@ -231,45 +162,50 @@ class cPluginLua :
 	template <typename FnT, typename... Args>
 	bool Call(FnT a_Fn, Args && ... a_Args)
 	{
-		cCSLock Lock(m_CriticalSection);
-		return m_LuaState.Call(a_Fn, a_Args...);
+		return cOperation(*this)().Call(a_Fn, a_Args...);
 	}
 
-	/** Adds the specified cResettable instance to m_Resettables, so that it is notified when the plugin is being closed. */
-	void AddResettable(cResettablePtr a_Resettable);
-
 protected:
-	/** Maps command name into Lua function reference */
-	typedef std::map<AString, int> CommandMap;
-
 	/** Provides an array of Lua function references */
-	typedef std::vector<cLuaState::cRef *> cLuaRefs;
+	typedef std::vector<cLuaState::cCallbackPtr> cLuaCallbacks;
 
 	/** Maps hook types into arrays of Lua function references to call for each hook type */
-	typedef std::map<int, cLuaRefs> cHookMap;
-
+	typedef std::map<int, cLuaCallbacks> cHookMap;
```

Comment:
Does this need to be a shared pointer? If a cLuaCallback outlives the hookmap, something is probably wrong.